### PR TITLE
fix: Limit scope of what's in a queue

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

File: packages/core/src/anchor/anchor-processing-loop.ts [hunk 1]
```diff
@@ -30,51 +30,54 @@ export class AnchorProcessingLoop {
     anchorStoreQueue: NamedTaskQueue
   ) {
     this.#anchorStoreQueue = anchorStoreQueue
-    this.#loop = new ProcessingLoop(logger, store.infiniteList(batchSize), (streamId) =>
-      this.#anchorStoreQueue.run(streamId.toString(), async () => {
```
Comment (ukstv) [inline packages/core/src/anchor/anchor-processing-loop.ts:34]: The big mutex gets downsized to a poor mans' version of Compare-and-Swap.

File: packages/core/src/anchor/anchor-processing-loop.ts [hunk 2]
```diff
@@ -30,51 +30,54 @@ export class AnchorProcessingLoop {
...
+        )
+        const isTerminal = await eventHandler.handle(event)
+        logger.debug(
+          `Anchor event with status ${event.status} for commit CID ${entry.cid} of Stream ${streamId} handled successfully`
+        )
+        if (isTerminal) {
+          logger.debug(`Removing entry from AnchorRequestStore for Stream ${streamId}`)
+          // Remove iff tip stored equals to the tip we processed
+          // Sort of Compare-and-Swap
+          await this.#anchorStoreQueue.run(streamId.toString(), async () => {
```
Comment (ukstv) [inline packages/core/src/anchor/anchor-processing-loop.ts:66]: We only need to guard a removal here.

**Review by stephhuynh18 — APPROVED** [review]
yahoo

File: packages/core/src/anchor/anchor-processing-loop.ts [hunk 3]
```diff
@@ -30,51 +30,54 @@ export class AnchorProcessingLoop {
...
+        const isTerminal = await eventHandler.handle(event)
+        logger.debug(
+          `Anchor event with status ${event.status} for commit CID ${entry.cid} of Stream ${streamId} handled successfully`
+        )
+        if (isTerminal) {
+          logger.debug(`Removing entry from AnchorRequestStore for Stream ${streamId}`)
+          // Remove iff tip stored equals to the tip we processed
+          // Sort of Compare-and-Swap.
+          await this.#anchorStoreQueue.run(streamId.toString(), async () => {
+            const loaded = await store.load(streamId)
```
Comment (stbrody) [inline packages/core/src/anchor/anchor-processing-loop.ts:67]: bummer about the extra state store read, but agreed this is the simplest fix without a transactional state store.  LGTM

Comment (ukstv) [reply packages/core/src/anchor/anchor-processing-loop.ts:67]: It is actually negligibly fast on qa-private node. Bummer, but a tolerable one.

**Pull request merged**